unit-tests/repository_test.py

```diff
@@ -90,18 +93,126 @@ def test_extract_from_http(self, repo: Tuple[Path, Path], tmp_path: Path):
         pass
 
 
+class TestGit:
+    """
+    Tests of the Git repository interface.
+    """
+    @fixture(scope='class')
+    def repo(self, tmp_path_factory: TempPathFactory) -> Tuple[Path, Path]:
+        """
+        Set up a repository and return its path along with the path of the
+        original file tree.
+        """
+        tree_path = tmp_path_factory.mktemp('tree', numbered=True)
+        (tree_path / 'alpha').write_text("First file")
+        (tree_path / 'beta').mkdir()
+        (tree_path / 'beta' / 'gamma').write_text("Second file")
+
+        repo_path = tmp_path_factory.mktemp('repo', numbered=True)
+        command = ['git', 'init', str(repo_path)]
+        assert run(command).returncode == 0
+        #
+        # We have to configure this information or the forthcoming commands
+        # will fail.
+        #
+        command = ['git', 'config', 'user.name', 'Testing Tester Tests']
+        assert run(command, cwd=str(repo_path)).returncode == 0
+        command = ['git', 'config', 'user.email', 'tester@example.com']
```

Comment:
I mentioned this too you offline - but I think the testing may need to restore the user's original settings or something because I think you have _become_ `Testing Tester Tests` since this part first ran under your account

Reply:
You are right and in a sphincter tightening manouver I updated those commits with the correct information. I'm not sure that a version control system should allow that sort of monkey business so easily but it was useful in this case.